vllm/utils.py

```diff
@@ -338,7 +338,26 @@ def create_kv_caches_with_random(
     return key_caches, value_caches
 
 
-class measure_cuda_memory:
+@lru_cache
+def print_warning_once(msg: str) -> None:
+    logger.warning(msg)
+
+
+def is_pin_memory_available() -> bool:
```

Comment:
QQ: Does it make sense to use cache for this function? Wondering if it helps performance.

Reply:
yes let me add that